frontend/src/container/APIKeys/APIKeys.tsx

```diff
@@ -0,0 +1,729 @@
+									<ClipboardEdit size={14} color={Color.BG_ROBIN_400} />
+								)}
+
+								{APIKey.role === USER_ROLES.VIEWER && (
+									<View size={14} color={Color.BG_ROBIN_400} />
+								)}
+
+								{!APIKey.role && <View size={14} color={Color.BG_ROBIN_400} />}
+							</div>
+
+							<div className="action-btn">
+								<PenLine size={14} onClick={(): void => showEditModal(APIKey)} />
+
+								<Trash2
+									size={14}
+									color={Color.BG_CHERRY_500}
+									onClick={(): void => showDeleteModal(APIKey)}
+								/>
+							</div>
+						</div>
+
+						<div className="api-key-details">
+							<div className="api-key-created-at">
+								<CalendarClock size={14} />
+								Last used <Minus size={12} />
+								<Typography.Text> {formattedDateAndTime} </Typography.Text>
+							</div>
+
+							{expiringInLessThan7Days && (
+								<div className="api-key-expires-in">
+									Expires in {APIKey.expiresAt} Days
+								</div>
+							)}
+						</div>
+					</div>
+				);
+			},
+		},
+	];
+
+	return (
+		<div className="api-key-container">
+			<div className="api-key-content">
+				<header>
+					<Typography.Title className="title">API Keys</Typography.Title>
+					<Typography.Text className="subtitle">
+						Create and manage access keys for the SigNoz API
+					</Typography.Text>
+				</header>
+
+				<div className="api-keys-search-add-new">
+					<Input
+						placeholder="Search for keys..."
+						prefix={<Search size={12} color={Color.BG_VANILLA_400} />}
+						value={searchValue}
+						onChange={handleSearch}
+					/>
+
+					<Button
+						className="add-new-api-key-btn"
+						type="primary"
+						onClick={showAddModal}
+					>
+						<Plus size={14} /> New Key
+					</Button>
+				</div>
+
+				<Table
+					columns={columns}
+					dataSource={dataSource}
+					loading={isLoading || isRefetching}
+					showHeader={false}
+					pagination={{
+						pageSize: 10,
+						showTotal: (total: number, range: number[]): string =>
+							`${range[0]}-${range[1]} of ${total} API Keys`,
+					}}
+				/>
+			</div>
+
+			{/* Delete Key Modal */}
+			<Modal
+				className="delete-api-key-modal"
+				title={<span className="title">Delete key</span>}
+				open={isDeleteModalOpen}
+				closable
+				afterClose={handleModalClose}
+				onCancel={hideDeleteViewModal}
+				destroyOnClose
+				footer={[
+					<Button
+						key="cancel"
+						onClick={hideDeleteViewModal}
+						className="cancel-btn"
+						icon={<X size={16} />}
+					>
+						Cancel
+					</Button>,
+					<Button
+						key="submit"
+						icon={<Trash2 size={16} />}
+						loading={isDeleteingAPIKey}
+						onClick={onDeleteHandler}
+						className="delete-btn"
+					>
+						Delete key
+					</Button>,
+				]}
+			>
+				<Typography.Text className="delete-text">
+					{t('delete_confirm_message', {
+						keyName: activeAPIKey?.name,
+					})}
+				</Typography.Text>
+			</Modal>
+
+			{/* Edit Key Modal */}
+			<Modal
```

Comment:
can extract the edit and add new code to a component and render the edit / add conditonally based on param

Reply:
The modal triggers happen via the listing page, so controlling the modals from the nested components will be back and forth between the child and parent. Hence this way.